webui/src/docs/rest-api.jsx

```diff
@@ -1,244 +1,540 @@
-		<p>Depending on the domain specification, other fields could be required in order to successfully publish a record.
-			The list of all the fields, with their description, multiplicity and controlled vocabulary,
-			is automatically returned to the user in case one of the required fields is missing.</p>
-
-		<Example>
-			{'curl -v -X PATCH -H \'Content-Type:application/json-patch+json\' \
-			-d \'[{"op": "add", "path":"/publication_state", "value": "published"}\' \
-			https://$HOSTNAME/api/records/$RECORD_ID/draft?access_token=$ACCESS_TOKEN'}
-		</Example>
-
-		<h3>Responses</h3>
-		<p>All response bodies are JSON encoded (UTF-8 encoded).</p>
-		<p>{'A record is represented as a JSON object: \u007B "field1": value, … \u007D '}</p>
-		<p>A collection of records is represented as a JSON array of objects:</p>
-		<p>{'[\u007B "field1": value, ... }, … ]\u007D'}</p>
-		<p>Timestamps are in UTC and formatted according to ISO 8601:</p>
-		<p>YYYY-MM-DDTHH:MM:SS+00:00</p>
+        <h1>The B2SHARE HTTP REST API</h1>
+        <p>The B2HARE HTTP REST API can be used for interacting with B2SHARE via
+            external services or applications, for example for integrating with
+            other web-sites (research community portals) or for uploading or
+            downloading large data sets that are not easily handled via a web
+            browser. This API can also be used for metadata harvesting.
+        </p>
+
+        <h3>Basic concepts</h3>
+        <p> A scientific <strong>community</strong> has
+            the roles of creating and maintaining metadata schemas and curating
+            the datasets which are part of a scientific domain or research project.
+            B2SHARE users can be part of one or more communities. Some selected
+            members of a community are also given the role of community
+            administrators, which grants them the special rights needed for the
+            schema definitions and record curation tasks.</p>
+        <p> Any user can upload scientific datasets into B2SHARE and thus
+            create data <strong>records</strong>. A record is comprised of data
+            files and associated metadata. The record’s <strong>metadata </strong>
+            consists of a set of common fixed metadata fields and a set of custom
+            metadata blocks, each block containing related metadata fields. A
+            record is always connected to one scientific community which has the
+            role of curating and maintaining it.</p>
+        <p> A data record can exist in several states. Immediately after creation a record
+            enters the 'draft' state. In this state the record is only accessible
+            by its owner and can be freely modified: its metadata can be changed
+            and files can be uploaded into or removed from it. A draft can be
+            published at any time, and through this action it changes its state
```

Comment:
Yes, we should document that, but after we have the feature. This doc should be in sync with the actual state of the product.